### ISSUE TITLE
Implement transducer for saved ginr DFA files

### ISSUE THREAD

**Issue by jrte** [jrte/ribose#17]
This fix will conclude issue #15. It will require refactoring `RuntimeModel` into a persistent store, a runtime component and a compiler component. The compiler component will use a ribose component to transduce ginr DFA files. There is a bit of vanity here, since the current compiler implementation is simple enough but RuntimeModel is too complex and need refactoring in any case.

```
clearHeader = clear[`~version`] clear[`~tapes`] clear[`~transitions`] clear[`~states`] clear[`~symbols`];
clearTransition = clear[`~from`] clear[`~to`] clear[`~tapes`] clear[`~length`] clear[`~symbol`];

Automaton = (
	(nil, clearHeader clearTransition select)
	('INR' digit+ @@ PasteAny) (tab, select[`~version`] cut select)
	(digit+ @@ PasteAny) (tab, select[`~tapes`] cut select)
	(digit+ @@ PasteAny) (tab, select[`~transitions`] cut select)
	(digit+ @@ PasteAny) (tab, select[`~states`] cut select)
	(digit+ @@ PasteAny) (tab, select[`~symbols`] cut select)
	(nl, header clearHeader)
	(
		(digit+ @@ PasteAny) (tab, select[`~from`] cut select)
		(digit+ @@ PasteAny) (tab, select[`~to`] cut select)
		(digit+ @@ PasteAny) (tab, select[`~tape`] cut select)
		(digit+ @@ PasteAny) (tab, select[`~length`] cut select count[`~length` `!eol`])
		(byte* @@ PasteBinary) (eol, select[`~symbol`] cut select)
		(nl, transition clearTransition)
	)*
	(eos, automaton)
):dfmain;

Automaton$(0,1 2):prsseq `build/ribose/automata/Automaton.pr`;
Automaton:save `build/ribose/automata/Automaton.dfa`;
```
Something like that...

**Comment by jrte** [comment]
This is almost complete, just needs some work in build.xml as of head @ 4626f81. The automaton transducer and target are implemented in [Automaton.inr](https://github.com/jrte/ribose/blob/head/ribose-patterns/Automaton.inr) and [AutomatonCompiler.java](https://github.com/jrte/ribose/blob/head/src/com/characterforming/jrte/engine/AutomatonCompiler.java).

CI build results look good:

```
Ribose (benchmarking fsts for extracting data from noisy inputs, vs similar regex, output muted):
  115080  2199300 20420730 kern-10.log
      LinuxKernelNil:  111 115 104 104 104 105 104 104 104 104 104 104 104 104 104 104 104 104 104 104 : 187.257 mb/s  45.875 nul/kb
               RegEx:  586 594 538 539 538 538 538 537 538 537 538 539 537 538 537 538 538 538 539 538 :  36.198 mb/s   5.173 ribose:regex
         LinuxKernel:  189 180 131 131 131 132 131 123 123 123 123 123 123 123 123 123 123 122 122 122 : 158.718 mb/s  45.875 nul/kb
               RegEx:  568 590 537 537 534 535 536 535 535 535 535 534 534 534 536 534 535 534 535 535 :  36.429 mb/s   4.357 ribose:regex
    LinuxKernelLoose:  496 249 181 180 182 180 180 182 179 179 179 182 179 179 180 184 180 179 180 183 : 107.893 mb/s  45.875 nul/kb
               RegEx:  574 598 547 549 546 545 546 546 546 545 545 546 547 545 546 545 545 544 546 546 :  35.701 mb/s   3.022 ribose:regex
   LinuxKernelStrict:  556 306 239 241 237 235 234 236 233 232 233 234 231 231 232 236 231 232 233 234 :  83.690 mb/s  45.875 nul/kb
               RegEx:  556 592 533 532 530 530 531 531 531 531 531 530 531 531 531 532 531 531 531 532 :  36.669 mb/s   2.282 ribose:regex
  23601  103261 1522902 verbosegc.vgc
          Tintervals:   33  19  19  11  81   8   8   9   9   8   9   9   8   8   9   8   8   9   8   9 : 170.865 mb/s  23.092 nul/kb
               RegEx:   23  12  11  11   6   6   6   5   5   4   4   4   4   4   4   4   4   4   4   4 : 363.088 mb/s   0.471 ribose:regex
          Sintervals:   33  13  80   8   8   9   8   8   9   8   8   8   8   8   8   9   8   8   8   8 : 179.303 mb/s  23.092 nul/kb
               RegEx:   24  11  12  12   6   6   5   6   7   6   6   4   4   4   4   5   4   4   4   4 : 337.756 mb/s   0.531 ribose:regex
Ribose (output equivalence tests):
 523  523 4303 build/patterns/test/Tintervals.jrte.out
 523  523 4303 build/patterns/test/Sintervals.jrte.out
 523  523 4303 build/patterns/test/Tintervals.regex.out
^^^ Identical
  70279  210837 7683656 build/patterns/test/LinuxKernel.regex.out
  70279  210837 7683656 build/patterns/test/LinuxKernel.jrte.out
  70279  210837 7683656 build/patterns/test/LinuxKernelLoose.jrte.out
  70279  210837 7683656 build/patterns/test/LinuxKernelStrict.jrte.out
```

**Comment by jrte** [comment]
This is complete in head branch and will be closed when head is merged into master.

**Comment by jrte** [comment]
Fixed in 1f47d51.
